src/goto-programs/validate_goto_model.cpp

```diff
@@ -0,0 +1,223 @@
+/*******************************************************************\
+Module: Validate Goto Programs
+
+Author: Diffblue
+
+Date: Oct 2018
+
+\*******************************************************************/
+
+#include "validate_goto_model.h"
+
+#include <set>
+
+#include <util/invariant.h>
+
+#include "goto_functions.h"
+
+namespace
+{
+class validate_goto_modelt
+{
+public:
+  using function_mapt = goto_functionst::function_mapt;
+
+  validate_goto_modelt(
+    const goto_functionst &goto_functions,
+    const validation_modet vm)
+    : vm{vm}, function_map{goto_functions.function_map}
+  {
+  }
+
+  void
+  do_goto_program_checks(goto_model_validation_optionst validation_options);
+
+private:
+  /// Check the goto_program has an entry point
+  ///
+  /// NB goto-cc -c will generate goto-programs without entry-points
```

Comment:
Super :pick: As a non-native speaker I find "NB" much harder to read than something like "note" or "note that", might be just me though

Reply:
n.b. / NB has the advantage of being universal (derives from latin), although a somewhat science / literary construct. 
Regardless, in this case it wasn't adding much, removed.